packages/@lwc/babel-plugin-component/src/component.ts

```diff
@@ -56,18 +81,18 @@ module.exports = function ({ types: t }) {
         let node = declarationPath.node;
 
         if (declarationPath.isClassDeclaration()) {
-            const hasIdentifier = t.isIdentifier(node.id);
+            const hasIdentifier = t.isIdentifier((node as types.ClassDeclaration).id);
             if (hasIdentifier) {
-                statementPath.insertBefore(node);
-                node = node.id;
+                statementPath!.insertBefore(node);
+                node = (node as types.ClassDeclaration).id;
             } else {
                 // if it does not have an id, we can treat it as a ClassExpression
-                t.toExpression(node);
+                t.toExpression(node as types.ClassDeclaration);
             }
         }
 
         return t.callExpression(registerComponentId, [
-            node,
+            node as unknown as types.Expression,
```

Comment:
This is a little surprising too. Is it because `node` might be of type `TSDeclareFunction`, which is not an `Expression`?

Reply:
Removed the `as unknown` cast. Still need a cast because of the way the variable is redefined, but I would prefer not to actually modify the code to fix it.